core/src/main/java/zipkin2/finagle/MutableSpan.java

```diff
@@ -111,22 +111,29 @@ synchronized MutableSpan setAddress(String key, InetSocketAddress ia) {
     return this;
   }
 
+  synchronized String getService() {
+    return service;
+  }
+
+  synchronized boolean isComplete() {
+    return isComplete;
+  }
+
   synchronized V1Span toSpan() {
     // fill in the host/service data for all the annotations
     for (V1Annotation a : annotations) {
       Endpoint ep = Endpoints.boundEndpoint(a.endpoint());
       span.addAnnotation(a.timestamp(), a.value(), ep.toBuilder().serviceName(service).build());
     }
     for (V1BinaryAnnotation ann : binaryAnnotations) {
-      if (ann.stringValue() == null) continue;
       Endpoint ep = Endpoints.boundEndpoint(ann.endpoint());
-      span.addBinaryAnnotation(
-          ann.key(), ann.stringValue(), ep.toBuilder().serviceName(service).build());
+      if (ann.stringValue() == null) { // service annotation
```

Comment:
This was a bug. We had no unit tests reading back the address details. Now we do